server/apikeys.go

```diff
@@ -0,0 +1,138 @@
+package main
+
+import (
+	"errors"
+	"log/slog"
+	"net/http"
+	"time"
+
+	"github.com/gin-gonic/gin"
+	"gorm.io/gorm"
+)
+
+type ApiKey struct {
+	ID        uint      `gorm:"primarykey" json:"id"`
+	CreatedAt time.Time `json:"createdAt"`
+	Value     string    `gorm:"not null" json:"value"`
+	UserID    uint      `gorm:"not null" json:"-"`
```

Comment:
We may want to make this a foreign key that maps to a User to ensure data integrity.